torch/_dynamo/bytecode_transformation.py

```diff
@@ -344,12 +345,26 @@ def transform_code_object(code, transformations, safe=False):
         "co_filename",
         "co_name",
         "co_firstlineno",
-        "co_lnotab",  # changed to "co_linetable" if python 3.10+
         "co_freevars",
         "co_cellvars",
     ]
-    if sys.version_info >= (3, 10):
-        keys = list(map(lambda x: x.replace("co_lnotab", "co_linetable"), keys))
+    if sys.version_info < (3, 8):
+        keys.insert(12, "co_lnotab")
+    elif sys.version_info < (3, 10):
+        keys.insert(1, "co_posonlyargcount")
+        keys.insert(13, "co_lnotab")
+    elif sys.version_info < (3, 11):
+        keys.insert(1, "co_posonlyargcount")
+        keys.insert(13, "co_linetable")
+    else:
+        # Python 3.11 changes to code keys are not fully documented.
+        # See https://github.com/python/cpython/blob/3.11/Objects/clinic/codeobject.c.h#L24
+        # for new format.
+        keys.insert(1, "co_posonlyargcount")
+        keys.insert(12, "co_qualname")
+        keys.insert(14, "co_linetable")
+        # not documented, but introduced in https://github.com/python/cpython/issues/84403
+        keys.insert(15, "co_exceptiontable")
```

Comment:
This is hard to read... perhaps replace this with:
```
keys = []
if sys.version_info < ...:
   keys.append(...)
keys.extend([...])
if sys.version_info < ...:
   keys.append(...)
keys.extend([...])

Reply:
Any reason this cannot be achieved by `dir(code)` ?

Reply:
`dir(code)` seems to give a list of attributes in order. This order is different than what the internal CPython code object expects.